extensions/amp-carousel/0.1/base-carousel.js

```diff
@@ -64,27 +65,20 @@ export class BaseCarousel extends AMP.BaseElement {
 
 
   buildButtons() {
-    this.prevButton_ = this.element.ownerDocument.createElement('div');
+    // TODO(aghassemi erwinm, #4146) i18n
+    this.prevButton_ = addScreenReaderButton(this.element,
```

Comment:
So for the carousel case where buttons are reused for both visible and invisible case, I rather take a different approach. Problem with current approach is knowledge of `i-amphtml-screen-reader` in carousel and the fact that `tabIndex` is now gone when controls are visible.

I suggest adding an additional `toggleScreenReaderButton(elem, isVisible)`. Carousel would build buttons as it did before. At the time when carousel wants to hide the buttons, it calls `toggleScreenReaderButton(buttonElem, false);` 

`toggleScreenReaderButton` needs to toggle `-amp-screen-reader` class, toggle tabIndex between `''` and `-1` and deal with toggleing `.left` between `''` and `2*count`. So some refactoring would be needed to share between `addScreenReaderButton` and `toggleScreenReaderButton`